mono/mini/method-to-ir.c

```diff
@@ -6996,8 +7019,25 @@ is_supported_tail_call (MonoCompile *cfg, MonoMethod *method, MonoMethod *cmetho
 		|| cfg->method->save_lmf
 		|| (cmethod->wrapper_type && cmethod->wrapper_type != MONO_WRAPPER_DYNAMIC_METHOD)
 		|| call_opcode == CEE_CALLI
-		|| ((virtual_ || call_opcode == CEE_CALLVIRT) && !cfg->backend->have_op_tail_call_membase)
-		|| vtable_arg // FIXME
+		|| (virtual_ && !cfg->backend->have_op_tail_call_membase)
+
+		// Passing vtable_arg uses (requires?) a volatile non-parameter register,
```

Comment:
It would probably be better to put these comments into some arm specific file like mini-arm.h.